File: src/main/java/at/yawk/fiction/impl/fimfiction/FimAuthentication.java
package at.yawk.fiction.impl.fimfiction;

import lombok.RequiredArgsConstructor;
import lombok.Value;

/**
 * @author yawkat
 */
@Value
@RequiredArgsConstructor
public class FimAuthentication {
    private final String username;
    private final String password;
    private final String otp;
    private final boolean remember;

    public FimAuthentication(String username, String password) {
        this(username, password, "", true);
    }
}

File: src/main/kotlin/at/yawk/fiction/impl/fimfiction/FimAuthentication.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package at.yawk.fiction.impl.fimfiction

/**
 * @author yawkat
 */
data class FimAuthentication(
        val username: String,
        val password: String,
        val otp: String,
        val isRemember: Boolean
) {
    constructor(username: String, password: String) : this(username, password, "", true)
}
